### ISSUE TITLE
after groupadmin add a user the user it does not show up at all users and when disabled it cannot be seen/recovered

### ISSUE THREAD

**Issue by JK74** [nextcloud/server#6217]
### Steps to reproduce
1. make a group
2. make a group admin
3. let the groupadmin share a folder to the group
4. the groupadmin now makes a new user1and adds it to his own user group
5. ISSUE1: THIS user1 is now visible to that group and can be found as well by other nextcloud users by the username but it does not show up in the users overview or can be find by the admin in his user overview.
6. now we disable user1 by the groupadmin of that group
7. ISSUE2: this user can be still be used anywhere as mentioned at item 5. And it probably is listed at the section 'disabled' users (uitgeschakeld in Dutch)  however there is no user visible there in the group disabled users although the count on the left side is '1' in that group
8. ISSUE3: how do we enable this user again as its now 'ghosted' in our database as the admin does not see it any longer neither can the groupadmin however people can still find it or share files to the user.
(by its username). 
9. The user cannot logon at all any longer to nextcloud as during logon is showed: 'User Disabled' at the logon screen.

Picture below shows:
group on the left 'Kerkenraad' what seems to have 3 users but only 3 are listed'.
![3userslisted](https://user-images.githubusercontent.com/17946255/29566793-515fcd52-874b-11e7-9a2a-e5e79ac50e1b.png)

Picture below shows 1 disabled user:
![enabled_users_3](https://user-images.githubusercontent.com/17946255/29566847-896fe31c-874b-11e7-9b5e-d56c3d5f7513.JPG)

Belows shows the section of disabled users where none are shown although the count is 1:
![disabled_users1](https://user-images.githubusercontent.com/17946255/29566841-8190e146-874b-11e7-8e8f-6daf571763d5.JPG)

### Expected behaviour
1. a user made by the group-admin should show up at all users sections to the nextcloud admin
2. if the disabled users section shows 1 then I would expect a username to show up on it on the right hand side
3. That group=admin or admin can re-enable an disabled user (now we can't see it anywhere)
4. admin should be able to see in his overview users created by groupadmins
### Actual behaviour
1. disabled user entry is 1 but no data behind it
2. disabled user is now entirely ghosted in nextcloud
3. admin does not have any overview on created or disabled users by groupadmin.

=====================================================================================

### General server configuration

**Operating system:** Linux nextcloud 4.4.0-92-generic #115-Ubuntu SMP Thu Aug 10 09:04:33 UTC 2017 x86_64

**Web server:** Apache/2.4.18 (Ubuntu) (apache2handler)

**Database:** mysql 5.7.19

**PHP version:** 7.0.22-0ubuntu0.16.04.1
<details>
        <summary>PHP-modules loaded</summary>

```
- Core
- date
- libxml
- openssl
- pcre
- zlib
- filter
- hash
- Reflection
- SPL
- session
- standard
- apache2handler
- mysqlnd
- PDO
- xml
- apcu
- calendar
- ctype
- curl
- dom
- mbstring
- fileinfo
- ftp
- gd
- gettext
- iconv
- igbinary
- json
- exif
- mysqli
- pdo_mysql
- Phar
- posix
- readline
- redis
- shmop
- SimpleXML
- sockets
- sysvmsg
- sysvsem
- sysvshm
- tokenizer
- wddx
- xmlreader
- xmlwriter
- xsl
- zip
- Zend OPcache
```
</details>

### Nextcloud configuration

**Nextcloud version:** 12.0.2 - 12.0.2.0

**Updated from an older Nextcloud/ownCloud or fresh install: from 11.0 > 11.x > 12.0 > 12.0.1 12.0.2

**Where did you install Nextcloud from:**  local storage

**Are you using external storage, if yes which one:** Array
(
    [0] => \OC\Files\Storage\Local
    [1] => \OCA\Files_External\Lib\Storage\FTP
    [2] => \OC\Files\Storage\DAV
    [3] => \OCA\Files_External\Lib\Storage\OwnCloud
    [4] => \OCA\Files_External\Lib\Storage\SFTP
    [5] => \OCA\Files_External\Lib\Storage\AmazonS3
    [6] => \OCA\Files_External\Lib\Storage\Dropbox
    [7] => \OCA\Files_External\Lib\Storage\Google
    [8] => \OCA\Files_External\Lib\Storage\Swift
    [9] => \OCA\Files_External\Lib\Storage\SFTP
    [10] => \OCA\Files_External\Lib\Storage\SMB
    [11] => \OCA\Files_External\Lib\Storage\SMB
)

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** YOUR ANSWER HERE (LDAP/ActiveDirectory/Webdav/...)

<details>
        <summary>Signing status</summary>

```
[]
```
</details>

<details>
        <summary>Enabled apps</summary>

```
- activity: 2.5.2
- admin_audit: 1.2.0
- admin_notifications: 1.0.0
- announcementcenter: 3.1.0
- bruteforcesettings: 1.0.2
- calendar: 1.5.3
- circles: 0.12.4
- comments: 1.2.0
- contacts: 1.5.3
- dav: 1.3.0
- external: 2.0.3
- federatedfilesharing: 1.2.0
- federation: 1.2.0
- files: 1.7.2
- files_automatedtagging: 1.2.2
- files_downloadactivity: 1.1.1
- files_external: 1.3.0
- files_pdfviewer: 1.1.1
- files_retention: 1.1.2
- files_sharing: 1.4.0
- files_texteditor: 2.4.1
- files_trashbin: 1.2.0
- files_versions: 1.5.0
- files_videoplayer: 1.1.0
- firstrunwizard: 2.1
- gallery: 17.0.0
- groupfolders: 1.1.0
- issuetemplate: 0.2.2
- logreader: 2.0.0
- lookup_server_connector: 1.0.0
- nextcloud_announcements: 1.1
- notes: 2.3.1
- notifications: 2.0.0
- oauth2: 1.0.5
- password_policy: 1.2.2
- provisioning_api: 1.2.0
- quota_warning: 1.0.1
- ransomware_protection: 1.0.1
- serverinfo: 1.2.0
- sharebymail: 1.2.0
- spreed: 2.0.1
- survey_client: 1.0.0
- systemtags: 1.2.0
- tasks: 0.9.5
- theming: 1.3.0
- twofactor_backupcodes: 1.1.1
- updatenotification: 1.2.0
- workflowengine: 1.2.0
```
</details>

<details>
        <summary>Disabled apps</summary>

```
- activitylog
- encryption
- files_antivirus
- user_external
- user_ldap
```
</details>

<details>
        <summary>Content of config/config.php</summary>

```
{
    "instanceid": "ocaacnf00r5p",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "xxx.xxx",
        "xxx.xx",
        "192.168.x.x",
        "192.168.x.x"
    ],
    "datadirectory": "\/xxx",
    "overwrite.cli.url": "https:\/\/xxx\/xxx",
    "dbtype": "mysql",
    "version": "12.0.2.0",
    "dbname": "xxx",
    "dbhost": "xxx",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "Europe\/Berlin",
    "installed": true,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "filelocking.enabled": "true",
    "redis": {
        "host": "\/var\/run\/redis\/redis.sock",
        "port": 0,
        "timeout": 0
    },
    "session_lifetime": 3600,
    "htaccess.RewriteBase": "\/nextcloud",
    "mail_smtpmode": "smtp",
    "mail_smtpsecure": "ssl",
    "mail_from_address": "noReply",
    "mail_domain": "xxx",
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtphost": "xxx",
    "mail_smtpport": "465",
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "loglevel": 0,
    "cron_log": true,
    "log_rotate_size": 104857600,
    "enable_previews": true,
    "enabledPreviewProviders": [
        "OC\\Preview\\PNG",
        "OC\\Preview\\JPEG",
        "OC\\Preview\\GIF",
        "OC\\Preview\\BMP",
        "OC\\Preview\\XBitmap",
        "OC\\Preview\\MarkDown",
        "OC\\Preview\\MP3",
        "OC\\Preview\\TXT",
        "OC\\Preview\\Movie"
    ],
    "preview_max_filesize_image": 200,
    "updater.secret": "***REMOVED SENSITIVE VALUE***",
    "maintenance": false,
    "theme": "",
    "singleuser": true,
    "updater.release.channel": "production"
}
```
</details>

### Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.101 Safari/537.36

**Operating system:** YOUR ANSWER HERE

### Logs

<details>
        <summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

<details>
        <summary>Nextcloud log (data/nextcloud.log)</summary>

```
Insert your Nextcloud log here
```
</details>

<details>
        <summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by JK74** [comment]
Interesting enough a few days later I looked again and now the user shows up at the disabled section. 
Is there a time-out or a certain background job doing this activity of showing disabled users? Meaning its not an instant available function?
If so:
1. what is the expected time-out that a user created by an groupadmin should show up to an admin
2. what is the exptected timeout that and disabled user shows up in the disabled user listing (so he can be made 'enabled' again?
thnx,

**Comment by MorrisJobke** [comment]
> ISSUE1: THIS user1 is now visible to that group and can be found as well by other nextcloud users by the username but it does not show up in the users overview or can be find by the admin in his user overview.

Fixed by #7292

**Comment by MorrisJobke** [comment]
Let's close then this here and tackle the user count with https://github.com/nextcloud/server/issues/8008
